--- gpac/Makefile.makefix	2010-05-05 09:18:35.000000000 -0600
+++ gpac/Makefile	2010-05-27 12:19:54.051316154 -0600
@@ -60,11 +60,11 @@
 	install -d "$(DESTDIR)$(mandir)"
 	install -d "$(DESTDIR)$(mandir)/man1"
 	if [ -d  doc ] ; then \
-	install -m 644 doc/man/mp4box.1 $(DESTDIR)$(mandir)/man1/ ; \
-	install -m 644 doc/man/mp4client.1 $(DESTDIR)$(mandir)/man1/ ; \
-	install -m 644 doc/man/gpac.1 $(DESTDIR)$(mandir)/man1/ ; \
+	install -pm 644 doc/man/mp4box.1 $(DESTDIR)$(mandir)/man1/ ; \
+	install -pm 644 doc/man/mp4client.1 $(DESTDIR)$(mandir)/man1/ ; \
+	install -pm 644 doc/man/gpac.1 $(DESTDIR)$(mandir)/man1/ ; \
 	install -d "$(DESTDIR)$(prefix)/share/gpac" ; \
-	install -m 644 doc/gpac.mp4 $(DESTDIR)$(prefix)/share/gpac/ ;  \
+	install -pm 644 doc/gpac.mp4 $(DESTDIR)$(prefix)/share/gpac/ ;  \
 	fi
 
 #	install -d "$(DESTDIR)$(prefix)/share/gpac/gui" ; \
@@ -97,23 +97,25 @@
 	install -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
 	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
 else
-	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac-$(VERSION).$(DYN_LIB_SUFFIX)
-	ln -sf libgpac-$(VERSION).$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
+	install $(INSTFLAGS) -m 755 bin/gcc/libgpac.$(DYN_LIB_SUFFIX) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX).$(VERSION)
+	ln -sf libgpac.$(DYN_LIB_SUFFIX).$(VERSION) $(DESTDIR)$(prefix)/$(libdir)/libgpac.$(DYN_LIB_SUFFIX)
+ifeq ($(DESTDIR)$(prefix),$(prefix))
 	ldconfig || true
 endif
 endif
+endif
 
 install-lib:
 	mkdir -p "$(DESTDIR)$(prefix)/include/gpac"
-	install -m 644 $(SRC_PATH)/include/gpac/*.h "$(DESTDIR)$(prefix)/include/gpac"
+	install -pm 644 $(SRC_PATH)/include/gpac/*.h "$(DESTDIR)$(prefix)/include/gpac"
 	mkdir -p "$(DESTDIR)$(prefix)/include/gpac/internal"
-	install -m 644 $(SRC_PATH)/include/gpac/internal/*.h "$(DESTDIR)$(prefix)/include/gpac/internal"
+	install -pm 644 $(SRC_PATH)/include/gpac/internal/*.h "$(DESTDIR)$(prefix)/include/gpac/internal"
 	mkdir -p "$(DESTDIR)$(prefix)/include/gpac/modules"
-	install -m 644 $(SRC_PATH)/include/gpac/modules/*.h "$(DESTDIR)$(prefix)/include/gpac/modules"
-	install -m 644 $(SRC_PATH)/config.h "$(DESTDIR)$(prefix)/include/gpac/configuration.h"
+	install -pm 644 $(SRC_PATH)/include/gpac/modules/*.h "$(DESTDIR)$(prefix)/include/gpac/modules"
+	install -pm 644 $(SRC_PATH)/config.h "$(DESTDIR)$(prefix)/include/gpac/configuration.h"
 ifeq ($(GPAC_ENST), yes)
 	mkdir -p "$(DESTDIR)$(prefix)/include/gpac/enst"
-	install -m 644 $(SRC_PATH)/include/gpac/enst/*.h "$(DESTDIR)$(prefix)/include/gpac/enst"
+	install -pm 644 $(SRC_PATH)/include/gpac/enst/*.h "$(DESTDIR)$(prefix)/include/gpac/enst"
 endif
 	mkdir -p "$(DESTDIR)$(prefix)/$(libdir)"
 	install -m 644 "./bin/gcc/libgpac_static.a" "$(DESTDIR)$(prefix)/$(libdir)"
