diff -up x264-stable-20110811/configure.nover x264-stable-20110811/configure
--- x264-stable-20110811/configure.nover	2011-08-11 20:21:38.000000000 +0200
+++ x264-stable-20110811/configure	2011-08-11 20:27:39.000000000 +0200
@@ -1057,7 +1057,7 @@ fi
 echo "LDFLAGSCLI = $LDFLAGSCLI" >> config.mak
 echo "CLI_LIBX264 = $CLI_LIBX264" >> config.mak
 
-${SRCPATH}/version.sh "${SRCPATH}" >> x264_config.h
+cat ${SRCPATH}/version.h >> ${SRCPATH}/x264_config.h
 
 pclibs="-L$libdir -lx264 $libpthread"
 
